homeassistant/components/advantage_air/__init__.py

```diff
@@ -53,29 +52,10 @@ async def async_get():
         update_interval=timedelta(seconds=ADVANTAGE_AIR_SYNC_INTERVAL),
     )
 
-    def error_handle_factory(func):
-        """Return the provided API function wrapped.
-
-        Adds an error handler and coordinator refresh.
-        """
-
-        async def error_handle(param):
-            try:
-                if await func(param):
-                    await coordinator.async_refresh()
-            except ApiError as err:
-                raise HomeAssistantError(err) from err
-
-        return error_handle
-
     await coordinator.async_config_entry_first_refresh()
 
     hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN][entry.entry_id] = {
-        "coordinator": coordinator,
-        "aircon": error_handle_factory(api.aircon.async_set),
-        "lights": error_handle_factory(api.lights.async_set),
-    }
+    hass.data[DOMAIN][entry.entry_id] = {"coordinator": coordinator, "api": api}
```

Comment:
In the future it would be nicer to store this as a dataclass since it makes it easier to type 

example in `lookin/models.py`

Reply:
Ill make that a seperate PR, but its a good idea.

Reply:
https://github.com/home-assistant/core/pull/91519
Based on this PR, so ill keep it in draft until this is merged.